src/Atmos/Model/AtmosModel.jl

```diff
@@ -86,7 +87,8 @@ function AtmosModel{FT}(::Type{AtmosLESConfiguration};
                          # TODO: Probably want to have different bc for state and diffusion...
                          boundarycondition::BC=NoFluxBC(),
                          init_state::IS=nothing,
-                         data_config::DC=nothing) where {FT<:AbstractFloat,O,RS,T,M,P,R,S,BC,IS,DC}
+                         data_config::DC=nothing,
+                         parameter_set::PS=nothing) where {FT<:AbstractFloat,O,RS,T,M,P,R,S,BC,IS,DC,PS}
```

Comment:
Why is this `nothing`? Shouldn't it default to the default ParameterSet?

Reply:
`ParameterSet` isn't defined yet.

```suggestion
                         parameter_set::PS) where {FT<:AbstractFloat,O,RS,T,M,P,R,S,BC,IS,DC,PS}
```

Reply:
Yes, but since this is included at the driver level, it basically must be given. I'll add an error checking--still working on this. This is going to be a big PR, since `parameter_set` will be needed virtually everywhere 😬

Reply:
I guess there's no simple way to ease into this.

Reply:
Actually, I'd like to split this PR into two steps/commits: 
 1) make the struct available everywhere that's needed (requires many interface changes, but guarantees no behavioral changes), also, passing in `parameter_set = nothing` is acceptable until step 2
 2) actually use the available data struct, and remove dependence on `PlanetParameters.jl`

Reply:
seems reasonable.